llvm/lib/Transforms/Utils/SimplifyIndVar.cpp

```diff
@@ -942,11 +956,11 @@ void SimplifyIndvar::simplifyUsers(PHINode *CurrIV, IVVisitor *V) {
     if ((isa<PtrToIntInst>(UseInst)) || (isa<TruncInst>(UseInst)))
       for (Use &U : UseInst->uses()) {
         Instruction *User = cast<Instruction>(U.getUser());
-        if (replaceIVUserWithLoopInvariant(User))
+        if (DT->dominates(L->getHeader(), User->getParent()) &&
```

Comment:
Why do we need this condition? Aren't you already checking it when pushing uses?

Reply:
We checked `UseInst` is dominated but not its users.

We need this condition to "protect" the following call `replaceIVUserWithLoopInvariant` to ensure we don't feed it with a phi that is already invariant. It was not necessary before because the loop is LCSSA and any exits would be fenced with phis.

I could have misunderstood your point though.

Reply:
I see, thanks for the explanation.